src/action.c

```diff
@@ -129,14 +133,18 @@ int main(int argc, char **argv) {
 
     free(target_pwd);
 
-    envar_set("SINGULARITY_CONTAINER", singularity_image_name(&image), 1); // Legacy PS1 support
-    envar_set("SINGULARITY_NAME", singularity_image_name(&image), 1);
-    envar_set("SINGULARITY_SHELL", singularity_registry_get("SHELL"), 1);
-    envar_set("SINGULARITY_APPNAME", singularity_registry_get("APPNAME"), 1);
-
     command = singularity_registry_get("COMMAND");
 
-    singularity_message(LOG, "USER=%s, IMAGE='%s', COMMAND='%s'\n", singularity_priv_getuser(), singularity_image_name(&image), singularity_registry_get("COMMAND"));
+    // don't use image object if DAEMON_JOIN is NULL because image object
```

Comment:
We will still need to open the image object, even though we don't mount it (unless I am missing something).